roofit/roofitcore/src/RooAddPdf.cxx

```diff
@@ -715,41 +715,31 @@ void RooAddPdf::updateCoefficients(CacheElem& cache, const RooArgSet* nset) cons
       for (auto coefArg : _coefList) {
         auto coef = static_cast<RooAbsReal*>(coefArg);
         myCoefCache[i] = coef->getVal(nset) ;
-        //cxcoutD(Caching) << "SYNC: orig coef[" << i << "] = " << myCoefCache[i] << endl ;
         lastCoef -= myCoefCache[i++];
       }			
       myCoefCache[_coefList.getSize()] = lastCoef ;
-      //cxcoutD(Caching) << "SYNC: orig coef[" << _coefList.getSize() << "] = " << myCoefCache[_coefList.getSize()] << endl ;
 
-
-      // Warn about coefficient degeneration
-      if ((lastCoef<-1e-05 || (lastCoef-1)>1e-5) && _coefErrCount-->0) {
-        coutW(Eval) << "RooAddPdf::updateCoefCache(" << GetName()
-		        << " WARNING: sum of PDF coefficients not in range [0-1], value="
-		        << 1-lastCoef ;
-        if (_coefErrCount==0) {
-          coutW(Eval) << " (no more will be printed)"  ;
+      // Treat coefficient degeneration
+      const float coefDegen = lastCoef < 0. ? -lastCoef : (lastCoef > 1. ? lastCoef - 1. : 0.);
+      if (coefDegen > 1.E-5) {
```

Comment:
I think the previous version of the check is much more quickly understood?

Reply:
You are right. What you don't see is that the next commits from the other branch use the excess below zero / above 1 for computations.